modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java

```diff
@@ -313,17 +313,17 @@ public static void deployWatchesAndUpdateMetaStorageRevision(MetaStorageManager
             throws NodeStoppingException, InterruptedException {
         // Watches are deployed before distributionZoneManager start in order to update Meta Storage revision before
         // distributionZoneManager's recovery.
-        metaStorageManager.deployWatches();
+        var deployWatchesFut = metaStorageManager.deployWatches();
 
         // Bump Meta Storage applied revision by modifying a fake key. DistributionZoneManager breaks on start if Vault is not empty, but
         // Meta Storage revision is equal to 0.
         var fakeKey = new ByteArray("foobar");
 
-        CompletableFuture<Boolean> invokeFuture = metaStorageManager.invoke(
+        CompletableFuture<Boolean> invokeFuture = deployWatchesFut.thenCompose(unused -> metaStorageManager.invoke(
```

Comment:
Is it the only place we should await deployWatches?

Reply:
I do not know. I add explicit waiting here because it is a common test class, which might be used in different circumstances.
Other places probably have internal guaranties or should wait to this future.

Reply:
I think, if before, we have no guarantees that MC start earlier than watches subscribe, it means thus tests are buggy.
Ok, add synchronous waiting in every invocation.

Reply:
> Ok, add synchronous waiting in every invocation.
Well, in tests I'd rather add assertThat(willSucceedFast()) with proper message in order to have a meaningful explanation if synchronous waiting will hang instead of simple join.